sympy/polys/matrices/eigen.py

```diff
@@ -32,7 +32,7 @@ def dom_eigenvects(A, l=Dummy('lambda')):
                 for i in range(rows)]
             EE = DomainMatrix(EE_items, (rows, cols), field)
 
-            basis = (A - EE).nullspace()
+            basis = (A - EE).nullspace(normalize=True)
```

Comment:
I have changed this to preserve existing behaviour since the default behaviour of `DomainMatrix.nullspace()` is changed here.

The default behaviour of `DomainMatrix.nullspace()` is now to *not* normalise by the bottom most entry unlike `Matrix`:
```python
In [7]: M = Matrix([[x, x+y], [x, x+y]])

In [8]: Matrix([[x, x+y], [x, x+y]]).nullspace()
Out[8]: 
⎡⎡     y⎤⎤
⎢⎢-1 - ─⎥⎥
⎢⎢     x⎥⎥
⎢⎢      ⎥⎥
⎣⎣  1   ⎦⎦

In [9]: Matrix([[x, x+y], [x, x+y]]).to_DM().nullspace().transpose().to_Matrix()
Out[9]: 
⎡-x - y⎤
⎢      ⎥
⎣  x   ⎦

In [11]: Matrix([[x, x+y], [x, x+y]]).to_DM().to_field().nullspace(normalize=True).transpose().to_Mat
    ...: rix()
Out[11]: 
⎡-x - y⎤
⎢──────⎥
⎢  x   ⎥
⎢      ⎥
⎣  1   ⎦
```
Not normalising is definitely better in terms of the algorithm but also for making subsequent use of the expressions. It is much better to have matrix with polynomial entries rather than a big denominator divided through.

We should change this code later and therefore change the output of `Matrix.eigenvects`. The intention in this PR though is just to add some new features and algorithms to `DomainMatrix` and not change any outwardly observable behaviour of `Matrix` (that will be done in a separate PR).